<?xml version="1.0"?>
<launch>
  <arg name="run_rqt_reconfigure" default="true" />
  <arg name="launch_phasespace_proxies" default="false" />


  <!-- Example launch file for the PhaseSpace ImpulseX2E motion-tracker
       with stylus interaction.

       This assumes that you have already started the PhaseSpace OWL
       server and the ROS phasespace node(s). You might also need to
       check the PhaseSpace alignment transform and publish that.
       Also, set the colormap_scale to a distance that is typical
       for the calibration errors in your setup; e.g. value=0.01
       results in full-scale blue/red for distance errors of 0.01m
       or 1 centimeter.
  -->


  <node pkg="tams_phasespace" name="phasespace_stylus" type="stylus_node" output="screen">
    <param name="debug_level" type="int" value="2" />
    <param name="parent_frame" type="string" value="world" />

    <param name="stylus_rigids_id" type="int" value="3" />
    <param name="stylus_driver_id" type="int" value="1356" />


    <param name="marker_radius" type="double" value="0.05" />
    <param name="colormap_scale" type="double" value="0.01" />
    <param name="colormap" type="int" value="0" />
    <param name="update_rate" type="double" value="1.0" />
    <param name="enabled_cameras" type="string" value="0,1,2,3,4,5,6,7" />

    <remap from="rigids" to="/phasespace_ros/rigids" />
    <remap from="buttons" to="/phasespace_ros/buttons" />
  </node>


  <group if="$(arg run_rqt_reconfigure)">
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" />
  </group>


  <!--
  <node pkg="tf" type="static_transform_publisher" name="world_broadcaster" 
        args="1 0 0 0 0 0 world phasespace 100" />
  -->

<!--  <arg unless="$(arg config)" name="command_args" value="" />
  <arg     if="$(arg config)" name="command_args" value="-d $(find taser_moveit)/launch/moveit.rviz" />
-->




  <!-- upload the F329 cage robot model to param server -->
  <include file="$(find tams_f329)/launch/f329_upload.launch" >
  </include>


  <!-- we do not have a robot connected, so publish fake joint states -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="/use_gui" value="true"/>
    <rosparam param="/source_list">[/config/fake_controller_joint_states]</rosparam>
  </node>


  <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="100.0" />
    <param name="tf_prefix" type="string" value="" />
  </node>


  <!-- stylus URDF model -->
  <param name="stylus_description" 
    command="$(find xacro)/xacro.py $(find tams_phasespace)/urdf/phasespace_stylus_instance.xacro" />


  <node name="stylus_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
     <param name="publish_frequency" type="double" value="100.0" />
     <remap from="robot_description" to="stylus_description" />
     <!-- <remap from="joint_states" to="stylus/joint_states" /> no non-fixed joints on the cube -->
  </node>


  <node pkg="tf" type="static_transform_publisher" name="lwr_apriltag_17_publisher" 
          args="0.6625 0.011 0.004   0 0 0   lwr_arm_base_link apriltag_17_link 10" />


  <!-- Phasespace tf and vizualization proxies -->
  <group if="$(arg launch_phasespace_proxies)">
    <include file="$(find phasespace_ros)/launch/proxies.launch" >
    </include>
  </group>


  <node name="tf_world_to_cage" pkg="tf" type="static_transform_publisher" 
        args="0 0 0 0 0 0 world cage_base_link 100" />


  <node name="tf_ps_base_to_ps" pkg="tf" type="static_transform_publisher" 
        args="0.8412099330 0.2694119958 -0.2766545397 -0.3797546300 0.5887960040 0.6027036213 -0.3819084073 cage_table_top_link phasespace 100" />


  <node name="tf_corner_to_corner" pkg="tf" type="static_transform_publisher" 
        args="0 0 0 3.1415 0 0 table_front_right_corner cage_table_top_link 100" />


  <node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen"
       args="$(find tams_phasespace)/config/stylus.rviz" />


</launch>
